lib/Wikia/src/Purger/CdnPurger.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Wikia\Purger;
+
+use VignetteRequest;
+use Wikia\Logger\Loggable;
+use Wikia\Rabbit\TaskProducer;
+use Wikia\Rabbit\TaskPublisher;
+
+/**
+ * Use Celery / RabbitMQ queue to send purge requests to Fastly
+ *
+ * This class will enqueue all URLs and send list of unique URLs at the end of the request handling
+ *
+ * Each service has different options so it must be a different task for each one.
+ * If there are going to be a lot of them, we should change the parameters to the celery worker
+ *
+ * @author Owen
+ * @author macbre
+ */
+class CdnPurger implements TaskProducer, Purger {
+
+	use Loggable;
+
+	const SERVICE_MEDIAWIKI = 'mediawiki';
+	const SERVICE_THUMBLR = 'thumblr.mediawiki';
+
+	/** @var array $buckets */
+	private $buckets = [
+		self::SERVICE_MEDIAWIKI => [
+			'urls' => [],
+			'keys' => [],
+		],
+		self::SERVICE_THUMBLR => [
+			'urls' => [],
+			'keys' => [],
+		],
+	];
```

Comment:
Perhaps we should rename this class to `CdnPurgerQueue`. It seems more descriptive to me.